tests/unit/parsers/test.manifestjson.js

```diff
@@ -5786,5 +5787,25 @@ describe('ManifestJSONParser', () => {
         ),
       ]);
     });
+
+    it('does not emit an error when data_collection_permissions is specified and the support is enabled', () => {
```

Comment:
followup: it would be good to add a few smoke tests that makes sure that the validation of the data_collection_permissions format is in place as expected, and one that confirms that setting data_collection_permissions to a truthy value that is not valid from a JSONSchema perspective doesn't get the missing data_collection_permissions error but it gets the schema validation error and the manifest to be considered invalid as expected.

note: as a side note, this PR is currently forked from the main branch before we imported the updated schema data that includes also the data_collection_permissions JSONSchema, and so to be able to add those tests we either need to do it on a rebased version of this PR or in a followup PR branched from a more recent tip.